pkg/disttask/framework/scheduler/scheduler.go

```diff
@@ -504,9 +504,10 @@ func (s *BaseScheduler) scheduleSubTask(
 	adjustedEligibleNodes := s.slotMgr.adjustEligibleNodes(eligibleNodes, task.Concurrency)
 	var size uint64
 	subTasks := make([]*proto.Subtask, 0, len(metas))
+	randPos := rand.Intn(len(adjustedEligibleNodes))
 	for i, meta := range metas {
-		// we assign the subtask to the instance in a round-robin way.
-		pos := i % len(adjustedEligibleNodes)
+		// we assign the subtask to the instance in a round-robin way at the random start position.
+		pos := (i + randPos) % len(adjustedEligibleNodes)
```

Comment:
we design it this way to pack subtasks onto as less number of nodes as possible, in this way we can make sure later task of higher concurrency can be run, to maximize resource usage

suppose we have 2 nodes, and task A/B of concurrency 4 each with one subtask, and task C of concurrency 8
currently, we will scheduler A/B on the first node, and C can run on the second
but with random node, A/B might scheduled on the same node, C cannot be run

Reply:
see https://github.com/pingcap/tidb/pull/59033

Reply:
Get it~, I will revert these part of code.